tests/ImageSharp.Tests/Formats/Qoi/QoiEncoderTests.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using SixLabors.ImageSharp.Formats.Qoi;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Tests.TestUtilities.ImageComparison;
+
+namespace SixLabors.ImageSharp.Tests.Formats.Qoi;
+
+[Trait("Format", "Qoi")]
+[ValidateDisposedMemoryAllocations]
+public class QoiEncoderTests
+{
+    [Theory]
+    [WithFile(TestImages.Qoi.Dice, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.EdgeCase, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Kodim10, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Kodim23, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.QoiLogo, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.TestCard, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.TestCardRGBA, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Wikipedia008, PixelTypes.Rgba32)]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("CodeQuality", "IDE0051:Quitar miembros privados no utilizados", Justification = "Function implicitly in tests")]
+    private static void Encode<TPixel>(TestImageProvider<TPixel> provider)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        using Image<TPixel> image = provider.GetImage();
+        using MemoryStream stream = new();
+        QoiEncoder encoder = new();
+        image.Save(stream, encoder);
+        stream.Position = 0;
+        using Image<TPixel> encodedImage = (Image<TPixel>)Image.Load(stream);
+        ImageComparingUtils.CompareWithReferenceDecoder(provider, encodedImage);
```

Comment:
We could avoid using our own decoder in this test if we do this instead:

```suggestion
        using Image<TPixel> image = provider.GetImage(new MagickReferenceDecoder());
        using MemoryStream stream = new();
        QoiEncoder encoder = new();
        image.Save(stream, encoder);
        stream.Position = 0;
        using Image<TPixel> encodedImage = (Image<TPixel>)Image.Load(stream);
        ImageComparer.Exact.CompareImages(image, encodedImage);
```

Cool that ImageMagick already supports QOI!

Reply:
I've just did it but it still passes all the tests, the wikipedia_008.qoi image doesn't throw an error

I'm doing this on windows, maybe OS is involved?

Reply:
It fails for me on Windows, and it also looks wrong with the viewer I linked in https://github.com/SixLabors/ImageSharp/pull/2446#discussion_r1246840717. Is it possible that the image you have locally differs from the one [you pushed to the repo](https://github.com/SixLabors/ImageSharp/blob/6245666d61cfc2d79981b81806b3d253d9bc8b07/tests/Images/Input/Qoi/wikipedia_008.qoi) for some reason?

Reply:
> Is it possible that the image you have locally differs from the one [you pushed to the repo](https://github.com/SixLabors/ImageSharp/blob/6245666d61cfc2d79981b81806b3d253d9bc8b07/tests/Images/Input/Qoi/wikipedia_008.qoi) for some reason?

Omg
I've just done some some tests and yeah, **it's malformed**
![imagen](https://github.com/SixLabors/ImageSharp/assets/17792760/708ef5ae-0907-40b8-837a-687dbb553049)

It's 120 bytes smaller than the original file, I'll search how do re-upload a file

Reply:
could be a https://github.com/SixLabors/ImageSharp/blob/main/.gitattributes thing...if git is treating like text then it might be trying to replace newlines etc.

Reply:
> could be a https://github.com/SixLabors/ImageSharp/blob/main/.gitattributes thing...if git is treating like text then it might be trying to replace newlines etc.

Thanks!
It looks like it's fixed now 😄

Reply:
You'll need to do this fix upstream at https://github.com/SixLabors/SharedInfrastructure

Once that is merged then you would update the submodule 

```
git submodule foreach git pull origin main
```

Otherwise, every time you run a local build the `.gitattributes` file is overwritten by the shared one.

Reply:
Opened https://github.com/SixLabors/SharedInfrastructure/pull/39.

Reply:
And merged.

Reply:
@LuisAlfredo92 You need to run this command now in your branch to update the submodule

https://github.com/SixLabors/ImageSharp/pull/2446#discussion_r1247237463